### ISSUE TITLE
Indentation Cycling with  else-if nested inside another if form not working correctly

### ISSUE THREAD

**Issue by indigo0086** [jcollard/elm-mode#90]
When attempting to do a nested else-if then form inside an if form the indentation adds extraneous spacing and you are unable to cycle proper indentation.  I use elm-format to correct it of course but the cycling seems to not be elm specific.  Asking around on twitter I was informed that it is using a haskell format instead of elm specific for tab cycling.  Here is a small recording ![peek 2016-12-17 13-38](https://cloud.githubusercontent.com/assets/478336/21289935/866cc3a2-c45e-11e6-9d6a-8377f9fa67b7.gif)

**Comment by mnannola** [comment]
I just ran into this as well, and not even on a nested if.

Here is the statement I have that causes an incorrect indentation on the second `else if` line

`
viewValidation model =
    let
        (color, message) =
            if model.password /= model.passwordAgain then
                ("red", "Passwords do not match!")
            else if String.length model.password < 9 then
                ("red", "Password needs to be longer than eight characters")
                    else if
            else
                ("green", "OK")
`

**Comment by huBIerTi2** [comment]
I have the same issue, hope it gets resolved.

**Comment by purcell** [comment]
Yeah, it's all a bit inscrutable to me, tbh, since it began as a mangled haskell indentation system.
